snuba/datasets/pluggable_entity.py

```diff
@@ -65,13 +67,20 @@ class PluggableEntity(Entity):
 
     def _get_builtin_validators(self) -> Sequence[QueryValidator]:
         mappers = [s.translation_mappers for s in self.storages]
-        return [
+        builtin_validators = [
             EntityContainsColumnsValidator(
                 EntityColumnSet(self.columns),
                 mappers,
                 self.validate_data_model or ColumnValidationMode.ERROR,
             )
         ]
+        enable_illegal_aggregate_validator = state.get_config(
+            "enable_illegal_aggregate_in_condition_validator", 0
+        )
+        assert enable_illegal_aggregate_validator is not None
+        if int(enable_illegal_aggregate_validator) == 1:
+            builtin_validators.append(IllegalAggregateInConditionValidator())
+        return builtin_validators
```

Comment:
I'd put the killswitch in the validator, not the entity code. That creates a separation of concerns where the entity is responsible for entity stuff and the processor is responsible for itself

Reply:
Makes sense. Updated!